framework/deproxy_server.py

```diff
@@ -0,0 +1,138 @@
+        try:
+            request = deproxy.Request(self.request_buffer)
+        except deproxy.IncompliteMessage:
+            return
+        except deproxy.ParseError:
+            tf_cfg.dbg(4, ('Deproxy: SrvConnection: Can\'t parse message\n'
+                           '<<<<<\n%s>>>>>'
+                           % self.request_buffer))
+        # Handler will be called even if buffer is empty.
+        if not self.request_buffer:
+            return
+        tf_cfg.dbg(4, '\tDeproxy: SrvConnection: Recieve request.')
+        tf_cfg.dbg(5, self.request_buffer)
+        response = self.server.recieve_request(request, self)
+        self.request_buffer = ''
+        if not response:
+            return
+        self.send_response(response)
+
+class BaseDeproxyServer(deproxy.Server):
+
+    def __init__(self, *args, **kwargs):
+        deproxy.Server.__init__(self, *args, **kwargs)
+        self.stop_procedures = [self.__stop_server]
+        self.is_polling = threading.Event()
+        self.sockets_changing = threading.Event()
+
+    def handle_accept(self):
+        pair = self.accept()
+        if pair is not None:
+            sock, _ = pair
+            handler = ServerConnection(server=self, sock=sock,
+                                       keep_alive=self.keep_alive)
+            self.connections.append(handler)
+            assert len(self.connections) <= self.conns_n, \
+                ('Too lot connections, expect %d, got %d'
+                 % (self.conns_n, len(self.connections)))
+
+    def run_start(self):
+        tf_cfg.dbg(3, '\tDeproxy: Server: Start on %s:%d.' % \
+                   (self.ip, self.port))
+
+        while self.is_polling.is_set():
+            pass
+        self.sockets_changing.set()
+
+        self.create_socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.set_reuse_addr()
+        self.bind((self.ip, self.port))
+        self.listen(socket.SOMAXCONN)
+
+        self.sockets_changing.clear()
+
+    def __stop_server(self):
+        tf_cfg.dbg(3, '\tDeproxy: Server: Stop on %s:%d.' % (self.ip,
+                                                             self.port))
+        while self.is_polling.is_set():
+            pass
+        self.sockets_changing.set()
+
+        self.close()
+        connections = [conn for conn in self.connections]
+        for conn in connections:
+            conn.handle_close()
+        if self.tester:
+            self.tester.servers.remove(self)
+
+        self.sockets_changing.clear()
+
+
+    def set_events(self, is_polling, sockets_changing):
+        self.is_polling = is_polling
+        self.sockets_changing = sockets_changing
+
+    @abc.abstractmethod
+    def recieve_request(self, request, connection):
+        raise NotImplementedError("Not implemented 'recieve_request()'")
+
+class StaticDeproxyServer(BaseDeproxyServer):
+
+    def __init__(self, *args, **kwargs):
+        self.response = kwargs['response']
```

Comment:
It saves only last for performance issues. Because I can configure wrk client and deproxy server, and saving all requests will be painful.

Reply:
But I'm thinking, how to make it better

Reply:
Why do you need to run `wrk` against `deproxy.Server`? it's very strange test case: wrk is a great benchmark and stress tool, while `deproxy.server` is very limited TCP server with poor performance. 
You technically can configure wrk against deproxy server but what benefit it can bring?

Reply:
For example i have nginx and depfoxy clients in group and i want to examine some requests, which tempesta do to backend

Reply:
S/clients/servers/

Reply:
Yes, in this case deproxy server will be overloaded but what about using separate server groups:
```
srv_group default {
<nginx><nginx><nginx>
}
srv_group slow {
<deproxy server>
}
http_sched_rules {
}
```
Say one header is different in wrk and deproxy client requests, e.g. Host:. then `http_sched_rules` may use it to send request to fast or slow server. In this way you can examine requests forwarded from tempesta without any performance issues.

Reply:
By the way, `ratio dinamic` or`ratio predict` schedulers may improve deproxy server performance. Deproxy is extremely slow compering to nginx, so dynamic schedulers will adjust rates of each backend, and slow deproxy will receive much less requets